systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java

```diff
@@ -0,0 +1,192 @@
+            buildImageOpenshift(name, namespace, dockerfilePath, imageTag, baseImage);
+        } else {
+            buildImageKaniko(name, namespace, dockerfilePath, imageTag, baseImage);
+        }
+    }
+
+    /**
+     * Build a specific image from passed Dockerfile and push it into internal registry.
+     * @param name Image name (it is also used as a name for all needed resources)
+     * @param namespace location where the build will happen
+     * @param dockerfilePath path to Dockerfile
+     * @param imageTag tag of the final image that will be pushed into internal registries
+     * @param baseImage used base image for the build
+     * @throws IOException
+     */
+    public static void buildImageKaniko(String name, String namespace, String dockerfilePath, String imageTag, String baseImage) throws IOException {
+        createDockerfileConfigMap(namespace, name, dockerfilePath);
+
+        Job kanikoJob = new JobBuilder()
+            .withNewMetadata()
+                .withName(name)
+                .withNamespace(namespace) // Change this to your namespace
+            .endMetadata()
+            .withNewSpec()
+                .withNewTemplate()
+                    .withNewMetadata()
+                        .withName(name)
+                    .endMetadata()
+                    .withNewSpec()
+                        .withRestartPolicy("Never")
+                        .withVolumes(
+                            new VolumeBuilder()
+                                    .withName(name)
+                                    .withNewConfigMap()
+                                        .withName(name)
+                                    .endConfigMap()
+                                    .build()
+                        )
+                        .addNewContainer()
+                            .withName(name)
+                            .withImage(KANIKO_IMAGE)
+                            .withArgs(
+                                "--dockerfile=/workspace/Dockerfile",
+                                "--destination=" + Environment.getImageOutputRegistry(namespace, name, imageTag),
+                                "--build-arg=BASE_IMAGE=" + baseImage,
+                                "--skip-tls-verify",
+                                "-v=debug")
+                            .withVolumeMounts(
+                                new VolumeMountBuilder()
+                                        .withName(name)
+                                        .withSubPath("Dockerfile")
+                                        .withMountPath("/workspace/Dockerfile")
+                                        .build()
+                            )
+                        .endContainer()
+                    .endSpec()
+                .endTemplate()
+            .endSpec()
+            .build();
+
+        ResourceManager.getInstance().createResourceWithoutWait(kanikoJob);
```

Comment:
Why it's without wait?

Reply:
It as just due to testing process. Updated to use wait method.